packages/client/src/__tests__/integration/integration-runner.test.ts

```diff
@@ -0,0 +1,94 @@
+import * as child_process from 'child_process'
+import * as fs from 'fs'
+import * as path from 'path'
+import * as util from 'util'
+
+const exec = util.promisify(child_process.exec)
+const wait = util.promisify(setTimeout)
+
+const MAX = 2
+const BACKOFF_MS = 100
+const excluded = ['']
+
+type TestType = 'happy' | 'errors'
+
+type Log = {
+  name: string
+  stderr?: string
+  stdout?: string
+}
+
+async function spawnJest(options: { dir: string; testType: TestType }): Promise<Log> {
+  const name = `${options.testType}/${options.dir}`
+
+  // Errors inside this child will propagate up
+  const child = await exec(`jest  ${name} --config=${__dirname.replace(/\\\\/g, '/')}/jest.config.js --runInBand`, {
```

Comment:
If this process errors the entire suite will fail fast, we could catch the error and propagate it like with the logs below. WDYT?